### ISSUE TITLE
Add a confirm option to CLI vm create commands which cause downtime

### ISSUE THREAD

**Issue by gbowerman** [Azure/azure-cli#10704]
Reference: Feature 4947992: [Linux E2E] Add a confirm option to CLI vm create commands which cause downtime

**Description:**
"If you az vm create with a size, say DS1_v3, and then repeat that same command with DS2_v3 or another size, it changes the VM size. (I was trying to create many instances and forgot to change the -n parameter)

It might be by design, but I wonder if we should warn customers and ask them to add a --yes or something, since it implies downtime. We get that info notice in the portal, not sure if CLI should emit some form of warning.

The RP probably can't do anything unless the "create or update" functionality is split in destructive vs. non-destructive updates. In EC2, there's a "modify instance attribute" method and the VM size can only be changed with a stopped VM."

We already have “az vm resize” so adding some kind of update parameter to destructive creates would make sense.

**Comment by yonzhan** [comment]
@qwordy  please take a look and comment.

**Comment by qwordy** [comment]
It's by design. It's called idempotence.
https://blogs.msdn.microsoft.com/cloud_solution_architect/2017/08/09/understanding-idempotence-and-scope-in-azure-resource-manager-templates/
But we'll discuss your advice.

**Comment by bureado** [comment]
I disagree that this behavior is a result of idempotent design. I think the concepts are orthogonal. 

I don't think anyone is disagreeing that issuing a second `az vm create` command with different parameters will result in a scoped deployment that is sufficiently different from the original one to mutate the state of the resource.

The point of this issue is that, unlike the example covered in the MSDN article which isn't disruptive of the VM operation, changing the VM size **automatically** reboots the instance with no warning to the user.

While the Compute RP _could_ change to accommodate this behavior (as it is done in other cloud providers) that's not what this issue is about. This is about issuing sufficient warning in `azure-cli` that such a specific change in the deployment (VM size) will reboot the instance.

Unlike an `az vm restart` operation, where the verb pretty much indicates what's going to happen, a user might not expect that reissuing `az vm create` (possibly because they forgot to change the VM name) will restart an existing VM.

**Comment by qwordy** [comment]
That makes sense. Changing the VM size automatically reboots the instance with no warning to the user. It's unexpected for some people. But it's hard to make a breaking change which adds a confirmation. I can add a warning first. And I will raise this issue to the team and have a discussion.
It's general problem of Azure CLI. We allow creating an existing resource again by create command, while the actual behavior is updating the existing resource.
